include/xtensor/xmath.hpp

```diff
@@ -3013,6 +3013,86 @@ namespace detail {
             return cov(eval(stack(xtuple(x, y))));
         }
     }
+
+    /*
+    * convolution mode placeholders
+    */
+    namespace conv_mode
+    {
+        struct valid{};
+        struct full {};
+    }
+
+    /*
+    * @brief computes the 1D convolution between two 1D expressions
+    *
+    * @param a 1D expression
+    * @param v 1D convolutional kernel
+    * @param mode placeholder from conv_mode namespace
+    * 
+    * @detail the algorithm convolves a with v and will incur a copy overhead
+    *   should v be longer than a.
+    */
+    template <class E1, class E2, class E3>
+    inline auto convolve(E1&& a, E2&& v, E3&& mode)
```

Comment:
E3 being the mode, it can be passed by value.

Reply:
Done!